origin-growth/src/resources/invite.js

```diff
@@ -55,6 +55,50 @@ class GrowthInvite {
     })
   }
 
+  /**
+   * Creates a referrer - referee connection in case this referee does not have a referrer yet
+   *
+   * @param {string} code - growth invitation code
+   */
+  static async makeReferralConnection(code, walletAddress) {
+    try {
+      const referralLink = await db.GrowthReferral.findOne({
+        where: {
+          refereeEthAddress: walletAddress
+        }
+      })
+      const referrer = await GrowthInvite._getReferrer(code)
+
+      console.log("Referral link: ", JSON.stringify(referralLink))
+      if (referralLink && referralLink.referrerEthAddress.toLowerCase() !== referrer.toLowerCase()) {
+        /* The referrer present in the referee's identity does not match
+         * with the referral data recorded in the DB.
+         * A corner case scenario this might happen is as follow:
+         *  - referee receives multiple invites.
+         *  - referee clicks on an invite and enrolls into growth campaing
+         *  - referee clicks on another invite link and enrolls again into
+         *  growth campaign.
+         *
+         * When this happens we ignore the subsequent invites and attribute all
+         * referees actions to the initial referrer.
+         *
+         */
+        logger.warn(`Referee ${walletAddress} already referred by ${referralLink.referrerEthAddress}`)
+        return
+      }
+
+      await db.GrowthReferral.create({
+        referrerEthAddress: referrer,
+        refereeEthAddress: walletAddress
```

Comment:
let's lower case referrer and walletAddress as we do in the rest of the code when writing any eth address to the DB

Reply:
yup thanks for catching this